phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/utils/CsvWriter.kt

```diff
@@ -0,0 +1,149 @@
+package fr.acinq.phoenix.utils
+
+import fr.acinq.phoenix.data.WalletPaymentInfo
+import fr.acinq.phoenix.utils.extensions.createdAt
+import kotlinx.datetime.Instant
+
+class CsvWriter {
+    
+    data class Configuration(
+        val includesFiat: Boolean,
+        val includesDescription: Boolean,
+        val includesNotes: Boolean
+    )
+
+    companion object {
+        private const val FIELD_ID          = "ID"
+        private const val FIELD_DATE        = "Date"
+        private const val FIELD_AMOUNT_BTC  = "Amount BTC"
+        private const val FIELD_AMOUNT_FIAT = "Amount Fiat"
+        private const val FIELD_DESCRIPTION = "Description"
+        private const val FIELD_NOTES       = "Notes"
```

Comment:
I think we need a few more rows:
- the direction of the payment (incoming or outgoing)
- the fee (in btc and fiat)
- for outgoing payments, a `destination` column? Depending on `OutgoingPayment.Details`, would contain a LN invoice or a Bitcoin address
- for incoming payments, an `origin` column? Depending on `IncomingPayment.Origin`, would contain a LN invoice or a Bitcoin address
- the ID is not really useful, as for outgoing payments it's an internal UUID, so I think we should use the payment hash instead (which is the ID for incoming payments)

Reply:
I think I should spell out the assumptions I'm making.

Most of the requests that we got for this feature mentioned they wanted it for tax purposes. So they probably want to import the CSV into tax software like [this](https://koinly.io/).

My experience with this kind of tax software, is that they support:
* services with API's that they can automatically import from (e.g. Coinbase)
* and for everything else, they support a CSV import

Oftentimes the instructions for the CSV import is something like:
* go to product X, and export your FULL transaction history
* go to page Y on our website, and click import

The other use case I had in mind was importing TX history into some kind of [money app](https://www.wiz.money/).

So for both of these, you're probably correct that the ID isn't really useful. We can drop that.

As for the `destination` and `origin` column, that's an interesting question...

So, for tax purposes, there's a big difference between:
- spending your bitcoin via an outgoing payment
- sending it to another wallet you own

One has tax consequences and the other doesn't. So putting information into these columns that helps to link "internal transfers" between wallets would be a good idea.

But I don't know if we need the entire LN invoice. What if we just put the payment hash (for LN payments), and the BTC address for swaps ?